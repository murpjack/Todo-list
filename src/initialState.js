const TODO_LIST = "TODO_LIST";

export const initialState = {
  list: localStorage.getItem(TODO_LIST) === null ? [
    {
      id: 0,
      name: "Write TODO list",
      isComplete: true
    },
    {
      id: 1,
      name: "看电影",
      isComplete: false
    },
    {
      id: 2,
      name: "Make something",
      isComplete: false
    }
  ] : JSON.parse(localStorage.getItem(TODO_LIST))
};
export default initialState;
