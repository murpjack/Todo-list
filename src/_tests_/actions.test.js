import React from "react";
import { shallow } from "enzyme";

import {
  addItem,
  removeItem,
  editItem,
  toggleItemComplete
} from "../actions/index";

import initialState from "../initialState";

describe("Test useReducer actions", () => {

  it("Test addItem action", () => {
    expect(true).toEqual(true);
  });


  it("Test removeItem action", () => {
    const {list} = initialState;
    const changedList = removeItem(list);
    // const startList = { name: "Write Todo list", isComplete: false };

    expect(changedList.length).toEqual(list.length - 1);
  });

  it("Test editItem action", () => {
    expect(true).toEqual(true);
  });

  it("Test toggleItemComplete action", () => {
    expect(true).toEqual(true);
  });
});
