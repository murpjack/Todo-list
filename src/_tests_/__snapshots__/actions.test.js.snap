// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ADD_ITEM payload output is correct 1`] = `
Object {
  "list": Array [
    Object {
      "id": 0,
      "isComplete": true,
      "name": "Write TODO list",
    },
    Object {
      "id": 1,
      "isComplete": false,
      "name": "看电影",
    },
    Object {
      "id": 2,
      "isComplete": false,
      "name": "Make something",
    },
    undefined,
  ],
}
`;

exports[`EDIT_ITEM payload output is correct 1`] = `
Object {
  "list": Array [
    Object {
      "id": 0,
      "isComplete": true,
      "name": "Write TODO list",
    },
    Object {
      "id": 1,
      "isComplete": false,
      "name": "看电影",
    },
    Object {
      "id": 2,
      "isComplete": false,
      "name": "Make something",
    },
  ],
}
`;

exports[`REMOVE_ITEM payload output is correct 1`] = `
Object {
  "list": Array [
    Object {
      "id": 0,
      "isComplete": true,
      "name": "Write TODO list",
    },
    Object {
      "id": 1,
      "isComplete": false,
      "name": "看电影",
    },
    Object {
      "id": 2,
      "isComplete": false,
      "name": "Make something",
    },
  ],
}
`;

exports[`TOGGLE_ITEM_COMPLETE payload output is correct 1`] = `
Object {
  "list": Array [
    Object {
      "id": 0,
      "isComplete": true,
      "name": "Write TODO list",
    },
    Object {
      "id": 1,
      "isComplete": false,
      "name": "看电影",
    },
    Object {
      "id": 2,
      "isComplete": false,
      "name": "Make something",
    },
  ],
}
`;
